lib/SIL/MemAccessUtils.cpp

```diff
@@ -21,6 +21,56 @@
 
 using namespace swift;
 
+SILValue swift::stripAccessMarkers(SILValue V) {
+  while (true) {
+    switch (V->getKind()) {
+    default:
+      return V;
+
+    case ValueKind::BeginBorrowInst:
+    case ValueKind::BeginAccessInst:
+      V = cast<SingleValueInstruction>(V)->getOperand(0);
+    }
+  }
+}
+
+// TODO: When the optimizer stops stripping begin_access markers, then we should
+// be able to assert that the result is a BeginAccessInst and the default case
+// is unreachable.
+SILValue swift::getAccessedAddress(SILValue v) {
+  assert(v->getType().isAddress());
+  while (true) {
+    switch (v->getKind()) {
+    default:
+      return v;
+
+    case ValueKind::BeginBorrowInst:
+    case ValueKind::StructElementAddrInst:
+    case ValueKind::TupleElementAddrInst:
+    case ValueKind::UncheckedTakeEnumDataAddrInst:
+    case ValueKind::TailAddrInst:
+    case ValueKind::IndexAddrInst:
+      v = cast<SingleValueInstruction>(v)->getOperand(0);
+      continue;
+    };
+  }
+}
+
+bool swift::isLetAddress(SILValue accessedAddress) {
+  assert(accessedAddress == getAccessedAddress(accessedAddress)
+         && "caller must find the address root");
+  // Is this an address of a "let" class member?
+  if (auto *REA = dyn_cast<RefElementAddrInst>(accessedAddress))
```

Comment:
Fix names here?

Reply:
I think you're talking about the remnants of capitalization. Sure I can fix that now.